lib/Model/Query.js

```diff
@@ -158,7 +158,7 @@ function Query(model, collectionName, expression, options) {
   // because otherwise maybeUnload would be looping through the entire results
   // set of each query on the same collection for every doc checked
   //
-  // Map of id -> true
+  // Map of id -> count of ids
   this.idMap = {};
```

Comment:
Actually, since idMap is now a number, we could use the check `idMap[id] > 0`, which might be better since it is more explicit.